Sources/GRPCProtobufCodeGen/ProtobufCodeGenParser.swift

```diff
@@ -98,9 +98,11 @@ package struct ProtobufCodeGenParser {
 
 extension ProtobufCodeGenParser {
   fileprivate func codeDependencies(file: FileDescriptor) -> [Dependency] {
-    var codeDependencies: [Dependency] = [
-      Dependency(module: "GRPCProtobuf", accessLevel: .internal)
-    ]
+    var codeDependencies: [Dependency] = []
+
+    if file.services.count > 0 {
+      codeDependencies.append(Dependency(module: "GRPCProtobuf", accessLevel: .internal))
+    }
```

Comment:
This isn't quite right: we'll still generate some imports in some cases unnecessarily (i.e. if the file defines a message which depends on a type which is bundled with SwiftProtobuf). We should return the empty array if there are no services.